File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.java
import java.util.Stack;
import java.util.Arrays;

class Solution {
    public int[] solution(int[] numbers) {
        int[] answer = new int[numbers.length];
        Arrays.fill(answer, -1);

        Stack<Integer> stack = new Stack<Integer>();
        stack.push(0);
        
        for(int i = 1; i<numbers.length; i++){
            while(!stack.isEmpty()){
                int idx = stack.pop();
                if(numbers[i] > numbers[idx]) answer[idx] = numbers[i];
                else {
                    stack.push(idx);
                    break;
                }
            }
            stack.push(i);
        }
        return answer;
    }
}

File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.js
function solution(numbers) {
    var answer = new Array(numbers.length).fill(-1);
    var stack = [0, ]
    
    for(let i = 1; i<numbers.length; i++) {
        while(stack) {
            let idx = stack.pop()
            
            if(numbers[idx] < numbers[i]) answer[idx] = numbers[i]
            else {
                stack.push(idx)
                break
            }
        }
        stack.push(i)
    }
    return answer;
}